typescript/schedules.md

```diff
@@ -0,0 +1,110 @@
+# Schedules
+
+- [Docs](https://docs.temporal.io/workflows/#schedules)
+- gRPC API:
+  - [methods](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/service.proto#L328-L370)
+  - [`request_response.proto`](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/request_response.proto#L821-L957)
+  - [`schedule/v1/message.proto`](https://github.com/temporalio/api/blob/master/temporal/api/schedule/v1/message.proto)
+  - [`ScheduleOverlapPolicy` enum](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/schedule.proto)
+
+## TS API
+
+```ts
+import { ScheduleClient, ScheduleOverlapPolicy } from '@temporalio/client'
+import { addWeeks } from 'date-fns'
+import { myWorkflow } from './workflows'
+
+const client = new ScheduleClient()
+
+const schedule = await client.create({
+  id: 'biz-id',
+  spec: {
+    // every hour at minute 5
+    interval: {
+      every: '1h',
+      at: '5m',
+    },
+    // every 20 days since epoch at day 2
+    // interval: {
+    //   every: '20d',
+    //   at: '2d'
+    // }
+    exclude: {
+      // skip 11:05 pm
+      hour: 23,
+      minute: 5,
+    },
+    endAt: addWeeks(new Date(), 4), 
+    jitter: '30s',
+    timezone: 'US/Eastern',
+  },
+  action: {
+    startWorkflow: {
+      workflowId: 'biz-id',
+      type: myWorkflow,
+      input: ['sorry this is the only thing reused, chad 😄'],
+    },
+  },
+  policies: {
+    overlap: ScheduleOverlapPolicy.BUFFER_ONE,
+    catchupWindow: '2m',
+    pauseOnFailure: true,
+  },
+  state: {
+    note: 'started schedule',
+    paused: true,
+    limitedActions: 10,
+  },
+  patch: {
+    triggerImmediately: true,
+    backfill: [
+      {
+        start: new Date(),
+        end: new Date(),
+        overlap: ScheduleOverlapPolicy.ALLOW_ALL,
+      },
+    ],
+    pause: true, // redundant with state.paused above (can use either)
+  },
+  memo,
+  searchAttributes,
+})
+
+const scheduleDescription = await schedule.describe()
+
+await schedule.listMatchingTimes({ start: new Date(), end: new Date() })
+
+await schedule.update({
+  spec,
+  action,
+  policies,
+  state,
+  conflictToken: scheduleDescription.conflictToken,
+})
+
+await schedule.patch({
+  triggerImmediately: true,
+  backfill,
+  unpause: true,
+})
+
+await schedule.delete()
+
+const { schedules, nextPageToken } = await client.list({
+  pageSize: 50,
+  nextPageToken: 'base64',
+})
```

Comment:
We can make this an async iterator so users don't have to paginate themselves.

Reply:
@dnr I thought schedule search attributes would be used for listing, doesn't look like this is the case. Is this coming?

@lorensr we should merge this with the `...MatchingTimes` API

Reply:
Merge how so? `listMatchingTimes` returns a list of Action start times

Reply:
@lorensr I think we should return schedule IDs here to avoid creating a handle for each schedule

Reply:
Updated my comment, I misunderstood initially.

Reply:
> I think we should return schedule IDs here to avoid creating a handle for each schedule

API doesn't give back ids, it gives back `ScheduleListEntry`s. Can update to make that clear and use async iterator:

```ts
for await (const schedule: ScheduleListEntry of client.list()) {
  const { id, memo, searchAttributes, info } = schedule
  // ...
}
```

Do we want to have both? Eg would be easier to use page version if coding the backend to a paginated UI.